main/lsp/requests/go_to_implementation.cc

```diff
@@ -0,0 +1,113 @@
+#include "main/lsp/requests/go_to_implementation.h"
+#include "core/lsp/QueryResponse.h"
+#include "main/lsp/json_types.h"
+#include "main/lsp/lsp.h"
+
+using namespace std;
+
+namespace sorbet::realmain::lsp {
+
+vector<core::SymbolRef> findChildren(const core::GlobalState &gs, core::SymbolRef ancestorClass) {
+    vector<core::SymbolRef> results;
+    auto used = gs.classAndModulesUsed();
+    for (u4 idx = 1; idx < used; idx++) {
+        core::SymbolRef ref(gs, core::SymbolRef::Kind::ClassOrModule, idx);
+        if (ref.data(gs)->derivesFrom(gs, ancestorClass)) {
+            results.push_back(ref);
+        }
+    }
+    return results;
+}
+
+variant<vector<core::Loc>, unique_ptr<ResponseError>> findMethodImplementations(const core::GlobalState &gs,
+                                                                                core::SymbolRef method) {
+    vector<core::Loc> locations;
+    auto owningClassSymbolRef = method.data(gs)->owner;
+    auto isAbstract = owningClassSymbolRef.data(gs)->isClassOrModuleAbstract();
```

Comment:
`owner` may not be a class or module, so you should check that before calling `isClassOrModuleAbstract`:

```suggestion
    auto isAbstract = owningClassSymbolRef.data(gs)->isClassOrModule() && owningClassSymbolRef.data(gs)->isClassOrModuleAbstract();
```